Init Alephium docs

DevRozaDev opened the pull request.

Sniezka1927 commented on docs/docs/alephium/collections.md:
```diff
@@ -0,0 +1,596 @@
+
+| Name  | Type    | Description           |
+| ----- | ------- | --------------------- |
+| caller   | Address | Address of the user who wants to know where the Token is stored. They pay the eventual fee.|
```
They won't know where it is stored using this function. `Caller` is requried to pay the init fee

Sniezka1927 commented on docs/docs/alephium/collections.md:
```diff
@@ -0,0 +1,596 @@
+
+| Name  | Type    | Description           |
+| ----- | ------- | --------------------- |
+| caller   | Address | Address of the user who wants to know where the Token is stored. They pay the eventual fee.|
```
It can be stored, or where it will be stored

Sniezka1927 commented on docs/docs/alephium/entrypoints.md:
````diff
@@ -0,0 +1,635 @@
+## Deployment
+
+```rust
+async function deployInvariant(
````
It is not contract entrypoint

Sniezka1927 commented on docs/docs/alephium/entrypoints.md:
````diff
@@ -0,0 +1,635 @@
+### Get protocol fee
+
+```rust
+pub fn Invariant::getProtocolFee() -> U256;
````
It shouldnt contain prefix
`pub fn getProtocolFee() -> U256;` is enough

Sniezka1927 commented on docs/docs/alephium/entrypoints.md:
````diff
@@ -0,0 +1,635 @@
+### Withdraw protocol fee
+
+```rust
+TxScript WithdrawProtocolFee(invariant: Invariant, poolKey: PoolKey);
````
I think it will be better to use same syntax for all entrypoints. 
`pub fn withdrawProtocolFee(poolKey: PoolKey) -> ();`

DevRozaDev commented on docs/docs/alephium/overview.md:
```diff
@@ -0,0 +1,140 @@
+---
+title: Overview
```
Good idea, I will think of a neat way of conveying Contract relationships

DevRozaDev commented:
https://github.com/invariant-labs/protocol/pull/370#discussion_r1675790978

addressed all the comments except this one, pushed the current state to make further feedback easier to leave

wojciech-cichocki commented on docs/docs/alephium/entrypoints.md:
````diff
@@ -0,0 +1,652 @@
+pub fn removePosition(index: U256) -> () 
+```
+
+This function removes a position from the user's position list and transfers the tokens used to create the position to the user's address.
````
It is worth adding information that all unclaimed fees generated by this position will also be transferred upon removing the position

wojciech-cichocki commented on docs/docs/alephium/entrypoints.md:
````diff
@@ -0,0 +1,652 @@
+pub fn getPosition(owner: Address, index: U256) -> (Bool, Position);
+```
+
+This function returns false as the first tuple variable if the position does not exist.
````
What is returned in the second field of the tuple?

wojciech-cichocki commented on docs/docs/alephium/entrypoints.md:
````diff
@@ -0,0 +1,652 @@
+) -> CalculateSwapResult
+```
+
+This function executes a swap based on the provided parameters. It transfers tokens from the user's address to the contract's address and vice versa. The swap will fail if the user does not have enough tokens, has not approved enough tokens, or if there is insufficient liquidity.
````
exceeding the swap limit is also failed case

wojciech-cichocki commented on docs/docs/alephium/entrypoints.md:
```diff
@@ -0,0 +1,652 @@
+| Name             | Type        | Description                                                                             |
+| ---------------- | ----------- | --------------------------------------------------------------------------------------- |
+| poolKey         | PoolKey     | Pool key of the pool on which you wish to perform the swap.                             |
+| xToY           | Bool        | Specifies the direction of the swap.                                                    |
```
It's worth adding what `true` and `false` mean here

wojciech-cichocki commented on docs/docs/alephium/entrypoints.md:
```diff
@@ -0,0 +1,652 @@
+| Code                | Description                                                                                                         |
+| ------------------- | ------------------------------------------------------------------------------------------------------------------- |
+| `ZeroAmount`      | Fails if the user attempts to perform a swap with zero amounts.                                                     |
+| `WrongPriceLimit` | Fails if the square root of price or price limit is incorrect. |
```
you can add that the limit direction was incorrectly specified and regardless of the swap outcome, it will be invalid due to this limit

wojciech-cichocki commented on docs/docs/alephium/entrypoints.md:
```diff
@@ -0,0 +1,652 @@
+| `NoGainSwap`        | Fails if the user would receive zero tokens.                                                                        |
+| `PoolNotFound`      | Fails if pool does not exist.                                                                                       |
+
+### Quote
```
Here, analogous changes to the "Swap" entrypoint section

wojciech-cichocki commented on docs/docs/alephium/entrypoints.md:
```diff
@@ -0,0 +1,652 @@
+| Name  | Type    | Description                                      |
+| ----- | ------- | ------------------------------------------------ |
+| poolKey   | PoolKey | A unique key that identifies the specified pool. |
+| index | I256     | The tick index in the tickmap.                   |
```
This is not the tick index in the tickmap, but the tick index (do not write this in the docs: this is tick index according to the formula `price = 1.0001^index`)

wojciech-cichocki commented on docs/docs/alephium/entrypoints.md:
```diff
@@ -0,0 +1,652 @@
+| Name  | Type    | Description                                      |
+| ----- | ------- | ------------------------------------------------ |
+| poolKey   | PoolKey | A unique key that identifies the specified pool. |
+| index | I256     | The tick index in the tickmap.                   |
```
incorrect description (similar as above comment)

wojciech-cichocki reviewed (changes requested):
There is a lack of mention regarding the `decimal types` used and a rough introduction to `U512` (how it is constructed, why such a decision was made)

DevRozaDev commented on docs/docs/alephium/collections.md:
```diff
@@ -0,0 +1,596 @@
+| ----- | ------- | --------------------- |
+| caller   | Address | Address of the user who wants to know where the Token is stored. They pay the eventual fee.|
+| reservePath | ByteVec     | Unique path. Required by current implementation, not used anywhere else.|
+| assetsToStore | U256     | Number of assets that will be stored right away.|
```
the assets are countable in this case, and that's why i'm using "number" instead of "amounts", the rest i agree on

DevRozaDev commented on docs/docs/alephium/entrypoints.md:
```diff
@@ -0,0 +1,652 @@
+| clamm | CLAMM | ContractId of the CLAMM module. |
+| reserveTemplateId | ByteVec | ContractId of a Reserve Contract. |
+| lastReserveId | ByteVec | ContractId of the same Reserve Contract. |
+| feeTierCount | U256 | Should be initialized to 0. |
```
that doesn't seem possible currently

wojciech-cichocki replied:
ok

DevRozaDev commented:
> There is a lack of mention regarding the `decimal types` used and a rough introduction to `U512` (how it is constructed, why such a decision was made)

We could add it in a separate PR unless it's necessary to have it in the initial version.

Sniezka1927 commented on docs/docs/alephium/storage.md:
```diff
@@ -0,0 +1,167 @@
+| Name     | Type      | Description                   |
+| -------- | --------- | ----------------------------- |
+| tokenX  | ByteVec | The address of x token.           |
+| tokenY  | ByteVec | The address of y token.           |
```
We are using contract id instead of address

Sniezka1927 commented on docs/docs/alephium/collections.md:
```diff
@@ -0,0 +1,595 @@
+
+Contract Invariant(...) extends Reserves(), ...{
+    ...
+        // Key: TokenId, Value: ReserveId
```
Either we mention the key and value in every place or nowhere

DevRozaDev replied:
Nowhere

Sniezka1927 commented on docs/docs/alephium/entrypoints.md:
````diff
@@ -0,0 +1,655 @@
+pub fn changeProtocolFee(newProtocolFee: U256) -> ();
+```
+
+This function allows for the adjustment of the current protocol fee percentage. Note that this operation is restricted to administrators.
````
The dex is designed to have single admin not mulitple administrators

Sniezka1927 reviewed (changes requested):
There are comments which are not pushed but they are marked as resolved.
The `Option` abstraction is also not mentioned

Sniezka1927 reviewed (approved):
The option abstraciton is not mentioned in the entrypoints section. I think may be worth to add it is own sub section.
Also it looks like it is not alligned with current protocol version. fee tiers are now static collection. If you like so you can apply these changes in the next pr
